server/client/js/main.js

```diff
@@ -6,64 +6,158 @@ let users = db.collection('users');
+                switch (error.code) {
+                    case "auth/invalid-email":
+                        alert("Invalid email.");
+                        break;
+                    case "auth/user-disabled":
+                        alert("User disabled.");
+                        break;
+                    case "auth/user-not-found":
+                        $("#signUpEmail").val(email);
+                        $("#signUpPassword").val(password);
+                        $("#signIn").hide();
+                        $("#signUp").show();
+                        break;
+                    case "auth/wrong-password":
+                        alert("Incorrect password.");
+                        break;
+                    default:
+                        alert(error.message);
+                }
+            });
+    });
+    $("#signInLink").click(() => {
+        $("#signUp").hide();
+        $("#signIn").show();
+
+    });
 
-// Create ephemeral user
-firebase.auth().signInAnonymously().catch((error) => {
-    console.log("error signing in anonymously:", error);
+    // Sign up logic
+    $("#signUpForm").submit(e => {
+        e.preventDefault();
+        let email = $("#signUpEmail").val();
+        let password = $("#signUpPassword").val();
+        let displayName = $("#signUpDisplayName").val();
+
+        auth.createUserWithEmailAndPassword(email, password)
+            .then(userCredential => {
+                userCredential.user.updateProfile({
+                    displayName: displayName
+                })
+                    .then(() => { 
+                        console.log("successfully set display name");
+                        $("#user").text(`${userCredential.user.displayName} (${userCredential.user.email})`);
+                        $("#content").show();
+                    })
+                    .catch(e => { console.log(error.message) });
+            })
+            .catch(error => {
+                switch (error.code) {
+                    case "auth/email-already-in-use":
+                        alert("Email already in use.");
+                        break;
+                    case "auth/invalid-email":
+                        alert("Invalid email.");
+                        break;
+                    case "auth/weak-password":
+                        alert("Password too weak.");
+                        break;
+                    default:
+                        alert(error.message);
+                }
+            });
+    });
+    $("#signUpLink").click(e => {
+        e.preventDefault();
+
+        $("#signIn").hide();
+        $("#signUp").show();
+    });
+
+    // Sign out logic
+    $("#signOut").click(e => {
+        e.preventDefault();
+
+        auth.signOut();
+        $("#content").hide();
+        $("#signIn").show();
+    });
 });
 
 // Once logged in
-firebase.auth().onAuthStateChanged((user) => {
+firebase.auth().onAuthStateChanged(user => {
```

Comment:
This function is fired on both initial page load, and after sign-in/sign-up.

The main things to know are:
- If `user` exists, a user is signed in. That may be as a result of the sign-in flow/sign-up flow during this page view, or from Firbase maintaining the login across page open/close.
- If `user` does not exist, a user is not signed in and the sign-in/up UI should be shown.